src/main.rs

```diff
@@ -103,55 +103,29 @@ fn main() -> crate::common::CliResult {
 
     let cli = match Cmd::try_parse() {
         Ok(cli) => cli,
-        Err(error) => {
-            match error.kind() {
-                clap::error::ErrorKind::DisplayHelp | clap::error::ErrorKind::DisplayVersion => {}
-                _ => match crate::js_command_match::JsCmd::try_parse() {
+        Err(error) => match error.kind() {
+            clap::error::ErrorKind::DisplayHelp | clap::error::ErrorKind::DisplayVersion => {
+                error.exit()
+            }
+            clap::error::ErrorKind::InvalidSubcommand => {
+                match crate::js_command_match::JsCmd::try_parse() {
                     Ok(js_cmd) => {
-                        match js_cmd.rust_command_generation() {
-                            Ok((vec_cmd, success_message)) => {
-                                eprintln!("{success_message}");
-                                eprintln!();
-                                eprintln!(
-                                    "    {}",
-                                    shell_words::join(
-                                        std::iter::once(near_cli_exec_path).chain(vec_cmd)
-                                    )
-                                    .yellow()
-                                );
-                            }
-                            Err(err) => {
-                                eprintln!("{}", err);
-                            }
-                        }
-                        std::process::exit(1);
+                        let vec_cmd = js_cmd.rust_command_generation();
+                        let cmd = &shell_words::join(
+                            std::iter::once(near_cli_exec_path.to_owned()).chain(vec_cmd),
+                        );
+                        Cmd::try_parse_from(cmd).unwrap()
```

Comment:
used `unwrap` here because it should always work

Reply:
You can replace `try_parse_from + unwrap` with just `parse_from`, if I remember correctly.

Reply:
apparently, `Cmd::parse_from`  does not exist, so I used `Parser::parse_from`